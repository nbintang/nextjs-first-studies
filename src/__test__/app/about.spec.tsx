import AboutPage from "@/app/about/page";
import { render, screen } from "@testing-library/react";

import '@testing-library/jest-dom';
import LayoutPage from "@/app/about/layout";

describe('About Page', () => { 
    it('should render', async () => {
        const page = render(
            <LayoutPage>
                <AboutPage/>
            </LayoutPage>
        );
        expect(page).toMatchSnapshot();
    });
});
