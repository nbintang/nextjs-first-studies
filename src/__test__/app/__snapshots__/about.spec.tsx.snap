// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="h-screen flex justify-center items-center"
      >
        <div
          class="flex gap-3 flex-row-reverse items-center"
        >
          <img
            alt="alt"
            class="rounded-md"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fwww.lineashoes.com%2Fmedia%2Fcatalog%2Fproduct%2Fcache%2F0cba5384d60be889ad311008a47fc53b%2F0%2F1%2F01-DR-MARTENS-FUH3YDRM0-DRM10085001-Black.jpg&w=640&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fwww.lineashoes.com%2Fmedia%2Fcatalog%2Fproduct%2Fcache%2F0cba5384d60be889ad311008a47fc53b%2F0%2F1%2F01-DR-MARTENS-FUH3YDRM0-DRM10085001-Black.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.lineashoes.com%2Fmedia%2Fcatalog%2Fproduct%2Fcache%2F0cba5384d60be889ad311008a47fc53b%2F0%2F1%2F01-DR-MARTENS-FUH3YDRM0-DRM10085001-Black.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="w-96"
          >
            <h1
              class="text-5xl font-semibold"
            >
              CUMAN BUAT LATIHAN
            </h1>
            <p>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium iste quaerat consectetur dignissimos hic cum at eius, itaque nemo temporibus accusamus ad aut? Accusantium dolore corrupti at veniam. Debitis, pariatur?
            </p>
            <div
              class="w-full flex justify-center"
            >
              <button
                class="bg-white text-black  rounded-md px-4 py-1 hover:opacity-90"
              >
                See Detail
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="h-screen flex justify-center items-center"
    >
      <div
        class="flex gap-3 flex-row-reverse items-center"
      >
        <img
          alt="alt"
          class="rounded-md"
          data-nimg="1"
          decoding="async"
          height="300"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fwww.lineashoes.com%2Fmedia%2Fcatalog%2Fproduct%2Fcache%2F0cba5384d60be889ad311008a47fc53b%2F0%2F1%2F01-DR-MARTENS-FUH3YDRM0-DRM10085001-Black.jpg&w=640&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fwww.lineashoes.com%2Fmedia%2Fcatalog%2Fproduct%2Fcache%2F0cba5384d60be889ad311008a47fc53b%2F0%2F1%2F01-DR-MARTENS-FUH3YDRM0-DRM10085001-Black.jpg&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fwww.lineashoes.com%2Fmedia%2Fcatalog%2Fproduct%2Fcache%2F0cba5384d60be889ad311008a47fc53b%2F0%2F1%2F01-DR-MARTENS-FUH3YDRM0-DRM10085001-Black.jpg&w=640&q=75 2x"
          style="color: transparent;"
          width="300"
        />
        <div
          class="w-96"
        >
          <h1
            class="text-5xl font-semibold"
          >
            CUMAN BUAT LATIHAN
          </h1>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusantium iste quaerat consectetur dignissimos hic cum at eius, itaque nemo temporibus accusamus ad aut? Accusantium dolore corrupti at veniam. Debitis, pariatur?
          </p>
          <div
            class="w-full flex justify-center"
          >
            <button
              class="bg-white text-black  rounded-md px-4 py-1 hover:opacity-90"
            >
              See Detail
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
